src/helics/core/CommsInterface.cpp

```diff
@@ -169,14 +235,17 @@ void CommsInterface::disconnect ()
     cnt = 0;
     while (tx_status.load () <= connection_status::connected)
     {
-        std::this_thread::sleep_for (std::chrono::milliseconds (50));
+        if (txTrigger.wait_for (std::chrono::milliseconds (800)))
+        {
+            continue;
+        }
         ++cnt;
-        if ((cnt & 31) == 0)
+        if ((cnt & 3) == 0)  // call this every 2400 milliseconds
         {
-            // try calling closeTransmitter again
+            // try calling closeReceiver again
             closeTransmitter ();
         }
-        if (cnt == 400)
+        if (cnt == 14)  // Eventually give up
```

Comment:
Not yet anyway.  I think we might expose a small number of them, since they are related to user actions in other federates. (like how long a federate waits to connect to a broker).  This one isn't as much, it is more about the OS, and in most cases just deals with closeout in a separate thread so it would only impact users on termination, and even in that case there is a check that will automatically terminate if the program is closing.  So I don't know that exposing this timeout would be that useful.    For this particular case, having a timeout is important, what its exact value is isn't as much.